`RestClientITCase` failed in CI, and is reproducible locally with `../../mvnw -Dnative=true clean install` (NB: IT) in `integration-tests/main` (after `../../mvnw clean install` in `core/deployment` on branch with this change)...

```
[ERROR] Failures: 
[ERROR]   RestClientITCase>RestClientTestCase.testMicroprofileCdiClientHeaderPassing:88 1 expectation failed.
JSON path header-name doesn't match.
Expected: some-not-at-all-random-header-value
  Actual: null

[ERROR]   RestClientITCase>RestClientTestCase.testMicroprofileClientHeaderPassing:98 1 expectation failed.
JSON path header-name doesn't match.
Expected: some-not-at-all-random-header-value
  Actual: null
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java
@@ -5,6 +5,8 @@ import static io.quarkus.deployment.util.ReflectUtil.toError;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.OutputStreamWriter;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.nio.charset.StandardCharsets;
@@ -278,12 +280,26 @@ public class ConfigurationSetup {
     private static <T> void withConverterHelper(final SmallRyeConfigBuilder builder, final Class<T> type, final int priority,
             final Class<? extends Converter<?>> converterClass) {
         try {
-            builder.withConverter(type, priority, ((Class<? extends Converter<T>>) converterClass).newInstance());
+            builder.withConverter(type, priority,
+                    ((Class<? extends Converter<T>>) converterClass).getDeclaredConstructor().newInstance());
         } catch (InstantiationException e) {
             throw toError(e);
         } catch (IllegalAccessException e) {
             throw toError(e);
+        } catch (NoSuchMethodException e) {
+            throw toError(e);
+        } catch (InvocationTargetException e) {
+            try {
+                throw e.getCause();
+            } catch (RuntimeException | Error e2) {
+                throw e2;
+            } catch (Throwable t) {
+                throw new UndeclaredThrowableException(t);
+            }
         }
+        // Constructor.newInstance() can also throw an IllegalArgumentException,
+        // or a SecurityException, both of which already are RuntimeException,
+        // so we do not catch those and just let them propagate.
     }
 
     /**
@@ -592,8 +608,9 @@ public class ConfigurationSetup {
             final StringBuilder methodName, final Map<String, MethodDescriptor> parseMethodCache) {
         final String methodNameStr = methodName.toString();
         final MethodDescriptor existing = parseMethodCache.get(methodNameStr);
-        if (existing != null)
+        if (existing != null) {
             return existing;
+        }
         try (MethodCreator body = cc.getMethodCreator(methodName.toString(), void.class, SmallRyeConfig.class,
                 NameIterator.class)) {
             body.setModifiers(Opcodes.ACC_PRIVATE | Opcodes.ACC_STATIC);

